### PR TITLE
Update logic behind polling & querying events

### LINKED ISSUE #8514: Events & actions huge load
## Bug description

If I try to visit Events & actions tab in posthog I can immediately see how my disk is trying to process it. After about 10 seconds I can see my events and it works as expected. And If I'm clicking on 'There are  new events. Click here to load them' posthog loads them instantly.

But somehow things changed and now after visiting Events tab posthog breaks. In my Clickhouse I can see that it still working on some query by disk loading (`iostat -4` for example). In Clickhouses query logs I can see strange query:

```
SELECT
    uuid,
    event,
    properties,
    timestamp,
    team_id,
    distinct_id,
    elements_chain,
    created_at
FROM
    posthog.events
where team_id = 1
AND timestamp > '2021-02-04 17:00:44.158000'AND timestamp < '2022-02-04 17:00:48.206451'
ORDER BY toDate(timestamp) DESC, timestamp DESC LIMIT 101
```
1. No space between conditions
2. Time range is a year. We have about 110 GB of data and it seems posthog can't wait much longer and stops working.

I've tried to upgrade to 1.32 version, but loading Events tab process still breaks with 502 error. Now I've downgraded posthog to 1.31 and in button 'Customize columns' I've pressed 'Reset to default' and it works well. If I will change the version to 1.32 - this error will remain.

## Expected behavior

Fields for selecting the time range in Events & actions tab. Set default value to 1 hour or something like that.

## How to reproduce

Not sure how to do it. In my case it happened randomly, but now repeats after upgrade to 1.32 

## Environment

Self-hosted PostHog (ClickHouse-based), version/commit: [release-1.31.1](https://hub.docker.com/layers/posthog/posthog/release-1.31.1/images/sha256-d825dd3eec25935089800d329eb6d82a77bcf565de8d24e8e4db1a1b036b6ab3?context=explore)

### PR DESCRIPTION
**Pull request by neilkakkar**
## Changes

Fixes #8514 #8187 

----

This PR gets rid of the concept of a `minimumQueryDate` for general querying, since it makes no sense to try and query all events over the past year, and then choose the limit: Columnar stores / Clickhouse don't work well with these kind of queries - there's no optimisation & clever stopping when you've seen `LIMIT` rows. 

Instead, we work with much smaller slices of time, that can go back to 1970, if the user so wishes. Instead of querying every year, we query in time slices of 5 days.

**Gotcha to think about**: If there are no events in between the oldest timestamp & 5 days before that, trying to load more doesn't work. Same for the case if you go to a persons tab to see their events, and there's nothing in the past 5 days = problem, we show empty results when we shouldn't!

To fix this, we have a failover to `minimumQueryDate`: If at any point, we query events and find nothing, we increase our search range to the maximum possible (assuming it wasn't already near the maximum).

(I need to write moaaar tests for this behaviour^, but opening this now so can get feedback on approach / other concerns I didn't think about)

And also, we'd probs need some tweaking around what's optimal here (and different failover levels if need be), but this seemed good enough for getting-into-production-and-testing.

----

Finally, I cleaned up tests a bit: matching get query strings was too brittle: if the after parameter goes before properties, it completely borks the tests. So, querying parameter objects instead now.

## How did you test this code?

logic tests and manual QA

### CODE REVIEW

File: frontend/src/scenes/events/eventsTableLogic.test.ts
Comment (pauldambra) [inline frontend/src/scenes/events/eventsTableLogic.test.ts:56]: this string manipulation feels like a signal that the `fromParamsGivenUrl` method needs a better API but no need to block this PR on that

File: frontend/src/scenes/events/eventsTableLogic.test.ts
Comment (pauldambra) [inline frontend/src/scenes/events/eventsTableLogic.test.ts:252]: much nicer tests for the URL!

File: frontend/src/scenes/events/eventsTableLogic.ts
Comment (pauldambra) [inline frontend/src/scenes/events/eventsTableLogic.ts:344]: this took some kreplits to parse, (if I'm right) if there are no API results and we're not close to the end of the pages of data in minimum query date range then load across the whole N months of minimum query date range to see if we get a result then... so there's a lot to unpack

I'd be tempted to give it a name or a comment to save someone else having to translate it in future

**Review by pauldambra — CHANGES_REQUESTED** [review]
one bit of clarification needed IMO

This looks like an awesome improvement! 🙌

File: frontend/src/scenes/events/eventsTableLogic.ts
Comment (neilkakkar) [reply frontend/src/scenes/events/eventsTableLogic.ts]: makes lots of sense, just realised this while reviewing myself, added a comment, lmk if that's sound / should change further!

File: frontend/src/scenes/events/eventsTableLogic.test.ts
Comment (neilkakkar) [inline frontend/src/scenes/events/eventsTableLogic.test.ts:417]: need one more test here to confirm before isn't changed, after becomes aYearAgo, when "load more" is clicked, but no events in past 5 days.

File: frontend/src/scenes/events/eventsTableLogic.ts
Comment (neilkakkar) [inline frontend/src/scenes/events/eventsTableLogic.ts:342]:
```suggestion
            // If no events, extend date range to maximum possible,
            // unless it was already at the maximum
```

**Review by pauldambra — APPROVED** [review]

**Review by mariusandra — CHANGES_REQUESTED** [review]
I think this is the right way to go, but some questions regarding coupling.

File: frontend/src/scenes/events/eventsTableLogic.ts
Comment (neilkakkar) [reply frontend/src/scenes/events/eventsTableLogic.ts:236]: Wanted to keep the "5 days is default time slice" logic in one place, selector seemed best.

Comment (neilkakkar) [reply frontend/src/scenes/events/eventsTableLogic.ts:236]: .. but could just be a standalone function now, since it's not using any other data anymore

File: frontend/src/scenes/events/eventsTableLogic.ts
Comment (neilkakkar) [reply frontend/src/scenes/events/eventsTableLogic.ts:346]: You mean send this value with the `fetchEvent` action? Hmm, nice, I like it better.

Comment (mariusandra) [reply frontend/src/scenes/events/eventsTableLogic.ts:346]: Hmm.. almost, but perhaps even better, can you just run

```ts
apiResponse = await api.get(`api/projects/${values.currentTeamId}/events/?${toParams(params)}`)
```

one more time with an updated `after` if the first query gives less than `$minimum` results? 

From the perspective of the rest of the app, it's weird that `fetchEvents` fires another `fetchEvents`, and I'm always afraid of runaway loops (aka white screens) with such logic. It should just handle everything internally in the listener/loader.

Comment (neilkakkar) [reply frontend/src/scenes/events/eventsTableLogic.ts]: Ah, I see. Sounds reasonable.

Comment (neilkakkar) [reply frontend/src/scenes/events/eventsTableLogic.ts]: It's a bit annoying to have to rewrite some of the fetching logic again, but can do, specially if a recursive action call in a listener breaks existing assumptions in app.

**Review by tiina303 — COMMENTED** [review]
Don't we have date and time filtering on properties now (https://posthog.slack.com/archives/C0113360FFV/p1644242202064449)? How does that work together with the 5 day limit?

File: frontend/src/scenes/events/eventsTableLogic.ts
Comment (neilkakkar) [reply frontend/src/scenes/events/eventsTableLogic.ts]: Great question! As long as there's some data, every 'load more' sends us back in time 5 days.

As soon as we don't find anymore data, we default to a date range of a year.

What you suggest would work as well, (and might be better for querying less), but I wanted to get the basic system in place first, see it works, and then address any new problems that might come up, if they do (like 'load more' taking too long, because we always default to a year, because no data in past 5 days 🤷 )

**Comment by neilkakkar** [comment]
Right on, had a convo with @mariusandra , and for now, also dropping time slices from 'load more events': `after` param is always 5 days ago, or a year ago.

**This means `load more` will be slower than it is today, after you've reached events from 5 days ago (since 2 requests for data).** But also simpler code.

Will revisit if people complain about load more taking too long, but good enough for first pass... (And I really want to get this in).

Will merge, unless objections :)

**Comment by mariusandra** [comment]
Do we care about the speed of "load more"? The task was to make sure the hot path (initial request scoped to a year), which is ~99% of the problems, gets handled. So 👍

**Comment by pauldambra** [comment]
If I'm doing it right I think this insight suggests we don't care https://app.posthog.com/insights/tuZ5eYkc

**Comment by neilkakkar** [comment]
Nice. Should've led with data as well, thanks for reminding. <3

7% absolute clicks, and [20% of people who actually click the button](https://app.posthog.com/insights/tuZ5eYkc/edit#filters=%7B%22insight%22%3A%22FUNNELS%22%2C%22date_from%22%3A%22-90d%22%2C%22actions%22%3A%5B%7B%22id%22%3A%2214833%22%2C%22name%22%3A%22Clicked%20load%20more%20events%22%2C%22type%22%3A%22actions%22%2C%22order%22%3A1%2C%22math%22%3A%22dau%22%7D%5D%2C%22events%22%3A%5B%7B%22id%22%3A%22%24pageview%22%2C%22name%22%3A%22%24pageview%22%2C%22type%22%3A%22events%22%2C%22order%22%3A0%2C%22properties%22%3A%5B%7B%22key%22%3A%22%24current_url%22%2C%22type%22%3A%22event%22%2C%22value%22%3A%22https%3A%2F%2Fapp.posthog.com%2Fevents%22%2C%22operator%22%3A%22icontains%22%7D%5D%2C%22math%22%3A%22dau%22%7D%5D%2C%22display%22%3A%22FunnelViz%22%2C%22interval%22%3A%22day%22%2C%22properties%22%3A%5B%5D%2C%22filter_test_accounts%22%3Atrue%2C%22funnel_viz_type%22%3A%22steps%22%2C%22exclusions%22%3A%5B%5D%7D):

I _feel_ that's a good enough chunk to care about, specially if `load more` being faster means more people use it, and for longer.

**Pull request merged**
